src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2907,7 +2907,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               // We're looking for a method (as indicated by FUNmode in the silent typed below),
               // so let's make sure our expected type is a MethodType
               val methArgs = NoSymbol.newSyntheticValueParams(argpts map { case NoType => WildcardType case tp => tp })
-              silent(_.typed(meth, mode.forFunMode, MethodType(methArgs, respt))) filter (isMonoContext) map { methTyped =>
+              silent(_.typed(meth.duplicate, mode.forFunMode, MethodType(methArgs, respt))) filter (isMonoContext) map { methTyped =>
```

Comment:
Although it might be preferable to leave this as `meth`, but change some code later (that does fallback typechecking) to perform `resetAttrs`. This pattern is a really fragile part of the typechecker, unfortunately.